data/tutorials/language/0lg_09_env.md

```diff
@@ -0,0 +1,675 @@
+
+For now, let's put aside those definitions and instead start playing with functions. Their meaning will arise from experience. Once things make sense, using these terms is just a means to interact with the community.
+
+This is a big takeaway. We believe functional programming (_ergo_ OCaml) is best understood by practising it rather than reading about it. Just like skateboarding, cooking, or woodworking. Learning by doing is not only possible, it is usually the easiest and the best way to start. Because in the end, the goal is not to be able to give the correct definition of abstract terms but to write OCaml code.
+
+## Applying Functions
+
+When several expressions are written side by side, the leftmost one should be a function. All the others are parameters. In OCaml, no parentheses are needed to express passing an argument to a function. Parentheses serve a single purpose: associating expressions to create subexpressions.
+```ocaml
+# max (21 * 2) (int_of_string "713");;
+- : int = 713
+```
+
+There are two alternative ways to apply functions: using the *application* `@@` operator or the *pipe* `|>` operator.
+```ocaml
+# sqrt 9.0;;
+- : float = 3.
+
+# sqrt @@ 9.0;;
+- : float = 3.
+```
+
+The `@@` application operator applies a parameter (on the right) to a function (on the left). It is useful when chaining several calls, as it avoids writing parentheses. Here is an example with and without parentheses:
+```ocaml
+# int_of_float (sqrt (float_of_int (int_of_string "81")));;
+- : int = 9
+
+# int_of_float @@ sqrt @@ float_of_int @@ int_of_string "81";;
+- : int = 9
+```
+
+The pipe operator (`|>`) also avoids parentheses but in reversed order: function on right, parameter on left.
+```ocaml
+# "81" |> int_of_string |> float_of_int |> sqrt |> truncate;;
+- : int = 9
+```
+This is just like a Unix shell pipe.
+
+## Anonymous Functions
+
+As citizens of the same level as other values, functions don't have to be bound to a name to exist (although some must, but this will be explained later). Function values not bound to names are called _anonymous functions_. Here are a couple of examples:
+```ocaml
+# fun x -> x;;
+- : 'a -> 'a = <fun>
+
+# fun x -> x * x;;
+- : int -> int = <fun>
+
+# fun s t -> s ^ " " ^ t ;;
+- : string -> string-> string = <fun>
+
+# function [] -> None | x :: _ -> Some x;;
+- : 'a list -> 'a option = <fun>
+```
+
+In order, here is what they are:
+- The identity function, takes anything and returns it unchanged.
+- The square function, takes an integer and returns it squared.
+- The function that takes two strings and returns their concatenation with a space character in between.
+- The function that takes a list that returns `None`, if the list is empty, and the tail of the list otherwise.
```

Comment:
```suggestion
In order, here is what they are:
- The identity function, which takes anything and returns it unchanged
- The square function, which takes an integer and returns it squared
- The function that takes two strings and returns their concatenation with a space character in between
- The function that takes a list that returns `None`, if the list is empty and the tail of the list otherwise
```